### PR TITLE
Add new check "implicit-flag-alias"

### LINKED ISSUE #8102: Feature request: implicit-flag-overlap
### Current problem

The following behaviour seemed confusing:

```py
>>> from enum import IntFlag
>>> class TestFlag(IntFlag):
...     A = 1
...     B = 2
...     C = 3
... 
>>> print([i for i in TestFlag])
[<TestFlag.A: 1>, <TestFlag.B: 2>]
```

(note that this is an `IntFlag`, not an `IntEnum`)

### Desired solution

The reason this happens is that `Flag` types are designed for setwise operations (bitwise operations, in the case of `IntFlag`), and, because `C` can be composed of `A` and `B`...

```py
>>> int(TestFlag.A | TestFlag.B)
3
>>> int(TestFlag.C)
3
```

...internally, `IntFlag` considers `C` to be an alias -- and doesn't include it when iterating over the members of the flag.

A `pylint` message something along the lines of `implicit-flag-overlap` could be emitted by a relevant checker when scanning the class declaration.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jayaddison**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |

## Description

This changeset implements an `implicit-flag-alias` check that inspects classes derived from `enum.Flag` and looks for member values that share common bit positions.

The commented sample code below may help explain:

```py
from enum import IntFlag

class Test(IntFlag):

    # valid cases:

    A = 1
    B = 2
    C = 2          # C is an alias (alternative name) for B, and is not highlighted by the check
    D = A | B      # D shares bit positions with A and B, but uses bitwise-operator syntax and is not highlighted

    # implicit-flag-alias cases:

    E = 3          # The value of E shares bit positions with A and B, so the check will highlight it
```

Closes #8102.
Replaces and continues from #8105 (origin branch accidentally deleted).

### CODE REVIEW

File: doc/data/messages/i/implicit-flag-alias/good.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from enum import IntFlag
+
+
+class FilePermissions(IntFlag):
+    READ = 1
+    WRITE = 2
+    EXECUTE = 4
```
Comment (jayaddison) [inline doc/data/messages/i/implicit-flag-alias/good.py:7]: Kind of a boring example.  More interesting suggestions welcome.

File: doc/data/messages/i/implicit-flag-alias/bad.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from enum import IntFlag
+
+
+class OverlappingFlags(IntFlag):
+    COLOUR = 1
+    SOUND = 2
+    SUBTITLES = 3  # [implicit-flag-alias]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/i/implicit-flag-alias/bad.py:7]:
```suggestion
class FilePermissions(IntFlag):
    READ = 1
    WRITE = 2
    EXECUTE = 3  # [implicit-flag-alias]
```

File: doc/data/messages/i/implicit-flag-alias/good.py (hunk 1 above)
Comment (Pierre-Sassoulas) [reply doc/data/messages/i/implicit-flag-alias/good.py:7]: It's nice and grounded in reality which help to understand 👍

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by jayaddison** [comment]
Thanks again @Pierre-Sassoulas :)

**Review by DanielNoord — APPROVED** [review]

**Comment by jayaddison** [comment]
Checking in to confirm that this looks like it's working as expected in v2.17.0 - I'll watch for any unexpected issues and/or false positives.

Tested with [v2.17.0](https://github.com/PyCQA/pylint/tree/2e78c414d24c9819936351c133b3edcad970abaa):

```py
# flag-overlap.py
from enum import IntFlag

class Chroma(IntFlag):
    BLEU = 0
    RED = 1
    BLEU = 2
    GRÜNE = 3
    黄色 = 4
```

```
$ pylint flag-overlap.py 
************* Module flag-overlap
flag-overlap.py:7:4: W0213: Flag member <Chroma.GRÜNE: 3> shares bit positions with <Chroma.RED: 1> (3 & 1 = 1), <Chroma.BLEU: 2> (3 & 2 = 2) (implicit-flag-alias)
flag-overlap.py:7:4: C2401: Attribute name "GRÜNE" contains a non-ASCII character, consider renaming it. (non-ascii-name)
flag-overlap.py:8:4: C2401: Attribute name "黄色" contains a non-ASCII character, consider renaming it. (non-ascii-name)

------------------------------------------------------------------
Your code has been rated at 5.71/10 (previous run: 7.14/10, -1.43)
```

**Pull request merged**